al_log_filter.js

```diff
@@ -38,10 +38,16 @@ var initJsonFilter = function(filter) {
  *  @NOTE: If json filter  is bad then 'messages' is returned unfiltered.
  */
 
-var filterJson = function(messages, filter) {
+var filterJson = function (messages, filter) {
     const filterJ = initJsonFilter(filter);
+    let result = filterItem(messages, filterJ);
```

Comment:
I think we need to move this line inside `if` statement down below, because `filterJ` can be `undefined`, basically
```
if (filterJ) {
  if (isArray(filterJ)) {
    filterJ.forEach();
  } else {
    return filterItem(messages, filterJ);
  }
}
```

Reply:
Done